src/hyperfine/export/mod.rs

```diff
@@ -0,0 +1,94 @@
+mod csv;
+
+use self::csv::CsvExporter;
+
+use std::io::Result;
+
+/// The ExportEntry is the main set of values that will
+/// be exported to files when requested.
+#[derive(Debug, Default, Clone, Serialize)]
+pub struct ExportEntry {
+    /// The command that was run
+    command: String,
+    /// The mean run time
+    mean: f64,
+    /// The standard deviation of all run times
+    stddev: f64,
+    /// Time spend in user space
+    user: f64,
+    /// Time spent in system space
+    system: f64,
+}
+
+impl ExportEntry {
+    /// Create a new ExportEntry with the given values
+    pub fn new(command: String, mean: f64, stddev: f64, user: f64, system: f64) -> Self {
+        ExportEntry {
+            command,
+            mean,
+            stddev,
+            user,
+            system,
+        }
+    }
+}
+
+/// The ResultExportType enum is used to denote the desired form
+/// of exporter to use for a given file.
+pub enum ResultExportType {
```

Comment:
Do we need this enum? `ResultExporter` is a trait, so we should be able to pass the trait objects directly.

Reply:
I wanted to take away all reasoning about what would be added as an exporter and leave it up to the manager to create the appropriate type

Reply:
Ok, sounds reasonable.